### ISSUE TITLE
ImageNet pre-trained weights for ResNet101

### ISSUE THREAD

**Issue by John1231983** [NanqingD/DeepLabV3-Tensorflow#8]
This is a discussion about using pre-train, not the bug

As you know, the Imagenet has 4 blocks, and the Imagenet pre-trained will store the weight of the 4 blocks. The size of these block will be

![image](https://user-images.githubusercontent.com/24875971/34238610-eb779ccc-e645-11e7-8d23-0ddcf8456642.png)

However, the deeplabv3 has 7 blocks (extra block 5, block 6, and block 7 as replicas of block 4). So if I understand correctly, it will be

`block3 (1x1,256; 3x3,256;1x1,1024)-> block 4 (1x1,512; 3x3,512;1x1,2048)-> block 5 (1x1,512; 3x3,512;1x1,2048)-> block 6 (1x1,512; 3x3,512;1x1,2048)-> block 7 (1x1,512; 3x3,512;1x1,2048)`

We can copy the weight of block4 to three extra blocks, except the first convolution of each extra block due to shape mismatch. The reason is that the first convolution of block 4 (conv1) will be `1x1x1024` (where 1024 is the last shape of input block 3), while the first convolution of the extra block is   `1x1x2048` (where 2048 is the last shape of input block 4). Do you have the same problem as above? Or the author just use pre-train weight for block 1-4, while block 5-7 are trained from scratch. Currently, I used `checkpoint_utils.init_from_checkpoint` to copy weight from block 4 to extra blocks.

**Comment by NanqingD** [comment]
@John1231983  Hi, John. Since now we only have 4 blocks, maybe we can incorporate Image-Net pre-trained weights. Could you make a pull request?

**Comment by John1231983** [comment]
Right. We can use imagnet pretrain now. However, I based on another implementation to develop deeplabv3 that uses low level tensorflow, instead of slim, thus it is very difficult to make a pull request for your code.

**Comment by John1231983** [comment]
This is my based implementation https://github.com/zhengyang-wang/Deeplab-v2--ResNet-101--Tensorflow

**Comment by NanqingD** [comment]
Hi, John, after taking your advice, I redo the training (this time train on 10582 and validate on 1449). The validation mIOU without multi-scale and random flip is 76.98%.
Thank you very much. I can't do this without your help.

If you are experienced with mulit-gpu training, welcome to contribute.

**Comment by John1231983** [comment]
Good job. I am training using pretrain imagnet but the result is not good now (about 70.8%). I guess the problem comes from training with batchnorm. When you train your model, do you aslo train the parameter of BN? The paper mentioned that he trained the BN with batchsize 16 for 30k. After that, he frozen the BN parameter and train with low learning rate
